helix-term/src/config.rs

```diff
@@ -55,6 +55,95 @@ impl Config {
     pub fn load_default() -> Result<Config, ConfigLoadError> {
         Config::load(helix_loader::config_file())
     }
+
+    // Load a merged config from configuration and $PWD/.helix/config.toml
+    pub fn load_merged_config() -> Config {
+        let root_config: Config = std::fs::read_to_string(helix_loader::config_file())
+            .ok()
+            .and_then(|config| toml::from_str(&config).ok())
+            .unwrap_or_else(|| {
+                eprintln!("Bad config: {:?}", helix_loader::config_file());
+                Config::halt_and_confirm("default");
+                Config::default()
+            });
+
+        // Load each config file
+        let local_config_values = helix_loader::local_config_dirs()
+            .into_iter()
+            .map(|path| path.join("config.toml"))
+            .chain([helix_loader::config_file()])
+            .filter_map(|file| Config::load_config_toml_values(&root_config, file));
+
+        // Merge configs and return, or alert user of error and load default
+        match local_config_values
+            .fold(toml::Value::Table(toml::value::Table::default()), |a, b| {
+                helix_loader::merge_toml_values(b, a, 3)
+            })
+            .try_into()
+        {
+            Ok(conf) => conf,
+            Err(_) => root_config,
+        }
+    }
+
+    // Load a specific config file if allowed by config
+    // Stay with toml::Values as they can be merged
+    pub fn load_config_toml_values(
+        root_config: &Config,
+        config_path: std::path::PathBuf,
+    ) -> Option<toml::Value> {
+        if config_path.exists() {
+            let mut confirmed = true;
+            if config_path != helix_loader::config_file() {
+                if root_config.editor.security.load_local_config {
+                    if root_config.editor.security.confirm_local_config {
+                        eprintln!(
+                            "Type yes<ENTER> to continue with loading config: {:#?}",
+                            config_path
+                        );
+                        let mut input = String::new();
+                        std::io::stdin().read_line(&mut input).unwrap_or_default();
```

Comment:
>  Then cargo run . will first ask you to type yes<RET> if you have a file, then open the file picker - here you can type. When you open a file, suddenly I cannot type any more.

Most likely what you mentioned is due to this line, `read_line` probably interfere with raw input mode or something, I am not sure.

Reply:
It turned out to be some setting in my config.toml
I should narrow it down (but I suspect it was my escape rebind)